Libraries/Image/AssetSourceResolver.js

```diff
@@ -81,7 +90,9 @@ class AssetSourceResolver {
         ? this.drawableFolderInBundle()
         : this.resourceIdentifierWithoutScale();
     } else {
-      return this.scaledAssetURLNearBundle();
+      return this.isCatalogAsset()
+        ? this.assetFromCatalog()
+        : this.scaledAssetURLNearBundle();
```

Comment:
To keep things open for other platforms we should do

```suggestion
return Platform.select({
 android: this.isLoadedFromFileSystem()
        ? this.drawableFolderInBundle()
        : this.resourceIdentifierWithoutScale(),
 ios: this.isCatalogAsset()
        ? this.assetFromCatalog()
        : this.scaledAssetURLNearBundle(),
  default: this.scaledAssetURLNearBundle()
  });
```